src/Core/Configurations/JsonConfigSchemaVaidator.cs

```diff
@@ -0,0 +1,147 @@
+    private static IFileSystem _fileSystem = new FileSystem();
+
+    /// <summary> 
+    /// Sets the logger and file system for the JSON config schema validator. 
+    /// </summary> 
+    /// <param name="jsonSchemaValidatorLogger">The logger to use for the JSON schema validator.</param> 
+    /// <param name="fileSystem">The file system to use for the JSON schema validator.</param>
+    public static void SetLoggerAndFileSystemForJsonConfigSchemaValidator(
+        ILogger<JsonSchemaValidator> jsonSchemaValidatorLogger,
+        IFileSystem fileSystem)
+    {
+        _logger = jsonSchemaValidatorLogger;
+        _fileSystem = fileSystem;
+    }
+
+     /// <summary> 
+    /// Validates a JSON schema against JSON data. 
+    /// </summary> 
+    /// <param name="jsonSchema">The JSON schema to validate against.</param> 
+    /// <param name="jsonData">The JSON data to validate.</param> 
+    /// <returns>A tuple containing a boolean indicating
+    /// if the validation was successful and a collection of validation errors if there were any.</returns> 
+    public static async Task<(bool ,ICollection<ValidationError>?)> ValidateJsonConfigWithSchemaAsync(string? jsonSchema, string? jsonData)
+    {
+        try{
+            JsonSchema schema = await JsonSchema.FromJsonAsync(jsonSchema);
+            ICollection<ValidationError> validationErrors = schema.Validate(jsonData, SchemaType.JsonSchema);
+
+            if (!validationErrors.Any())
+            {
+                _logger!.LogInformation("The config satisfies the schema requirements.");
+                return (true, null);
+            }
+            else
+            {
+                return (false, validationErrors);
+            }
+        }
+        catch (Exception e)
+        {
+            _logger!.LogError($"Failed to validate config against schema due to \n{e.Message}");
+            return (false, null);
+        }
+    }
+
+    /// <summary> 
+    /// Validates the given json config. 
+    /// </summary> 
+    /// <param name="jsonConfig">The JSON config file to validate.</param> 
+    public static async Task ValidateJsonConfig(string jsonConfig)
+    {
+        // deserialize jsonData to runtimeConfig
+        if (!RuntimeConfigLoader.TryParseConfig(jsonConfig, out RuntimeConfig? config))
+        {
+            Console.WriteLine("The config is invalid.");
+            return;
+        }
+        
+
+        string? jsonSchema = await GetJsonSchema(config);
+
+        if (string.IsNullOrWhiteSpace(jsonSchema))
+        {
+            _logger!.LogWarning("The schema file is invalid. Unable to verify provided config against schema.");
+            return;
+        }
+
+        (bool isValid, ICollection<ValidationError>? validationErrors) = await ValidateJsonConfigWithSchemaAsync(jsonSchema, jsonConfig);
+        
+        if (!isValid && validationErrors != null)
```

Comment:
Curious why only a check on `isValid` is not sufficient here

Reply:
for cases where it failed to validate the schema due to exception caused while validating or parsing the schema